compiler/src/main/java/com/juul/pommel/compiler/PommelWriter.kt

```diff
@@ -44,28 +45,48 @@ class PommelWriter(
                 }
             }
             .addMethod(
-                MethodSpec.methodBuilder(targetType.rawClassName().provideFunctionName())
-                    .addAnnotation(PROVIDES)
-                    .apply { scope?.let { addAnnotation(it) } }
-                    .addModifiers(Modifier.PUBLIC)
-                    .returns(binds)
-                    .applyEach(parameters) {
-                        addParameter(it.qualifiedType, it.simpleName.toString())
-                    }
-                    .addStatement(
-                        "return new \$T(\n\$L)", targetType,
-                        parameters.map { CodeBlock.of("\$N", it.simpleName) }.joinToCode(",\n")
-                    )
-                    .build()
+                if (targetType == binds) {
+                    writeProvidesMethod()
+                } else {
+                    writeBindsMethod()
+                }
             )
             .build()
     }
+
+    private fun writeProvidesMethod(): MethodSpec {
+        return MethodSpec.methodBuilder(targetType.rawClassName().provideFunctionName())
+            .addAnnotation(PROVIDES)
+            .apply { scope?.let { addAnnotation(it) } }
+            .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+            .returns(binds)
+            .applyEach(parameters) {
+                addParameter(it.qualifiedType, it.simpleName.toString())
+            }
+            .addStatement(
+                "return new \$T(\n\$L)", targetType,
+                parameters.map { CodeBlock.of("\$N", it.simpleName) }.joinToCode(",\n")
+            )
+            .build()
+    }
+
+    private fun writeBindsMethod(): MethodSpec {
+        return MethodSpec.methodBuilder(targetType.rawClassName().bindsFunctionName())
+            .addAnnotation(BINDS)
+            .apply { scope?.let { addAnnotation(it) } }
```

Comment:
These nested scoping functions make this harder to read than it needs to be (IMHO).

```suggestion
            .apply { if (scope != null) addAnnotation(scope) }
```

If you agree, looks like it's repeated a few times in this PR.

Reply:
I like the scope functions lol but I can see how it can be hard to read and not immediately clear to some reading the code what is going on. I'll make the change